services/ui-src/src/components/fields/ChoiceListField.tsx

```diff
@@ -67,23 +64,10 @@ export const ChoiceListField = ({
     if (displayValue) {
       form.setValue(name, displayValue, { shouldValidate: true });
       // update DOM choices checked status
-      clearChildren(choices);
+      clearUncheckedNestedFields(choices);
     }
   }, [displayValue]);
 
-  const autosaveSelections = async (selectedOptions: Choice[] | null) => {
-    const initialDataToSend: any = { [name]: selectedOptions };
-    const cleanedDataToSend = clearChildren(choices, initialDataToSend);
-
-    const reportKeys = createReportKeys(report?.id, state);
-    const dataToWrite = createDataToWrite(
-      { [name]: cleanedDataToSend },
-      full_name
-    );
-
-    await updateReport(reportKeys, dataToWrite);
-  };
```

Comment:
Note: I don't think we'll need this anymore because of the new `getNestedFields` function.